fix(deps): restore functionality of USE_EXISTING_SRC_DIR

jamessan opened the pull request:
30a0299bc removed the USE_EXISTING_SRC_DIR hack which broke building the
nightly PPA since ExternalProject tried to download the sources.

dundargoc commented:
I think 30a0299bc6d1ceedd04d897cf56b298dd3ded0cd is the offending commit.

Do you mind adding a job in `.github/workflow/build.yml` that emulates the PPA build to prevent this in the future?

jamessan commented:
> Do you mind adding a job in `.github/workflow/build.yml` that emulates the PPA build to prevent this in the future?

Do you want a completely separate job or add a couple steps to the old cmake job?

jamessan committed:
ci: add check to ensure USE_EXISTING_SRC_DIR=ON builds work

jamessan committed:
fix(deps): restore functionality of USE_EXISTING_SRC_DIR

30a0299bc removed the USE_EXISTING_SRC_DIR hack which broke building the
nightly PPA since ExternalProject tried to download the sources.

dundargoc commented on .github/workflows/build.yml:
```diff
@@ -78,3 +78,30 @@ jobs:
+          cd .deps/build/src
+          rm -rf ./*-build
+          rm -rf ./*-stamp/*-{configure,build,install,done}
+          for d in *; do (cd "$d"; rm -rf ./autom4te.cache; make clean || true; make distclean || true); done
```
nitpick, not a blocker: is this necessary? The only files that should remain are the source files after the externalproject `PATCH_COMMAND`, so nothing should've been built I feel like. Not important though.

dundargoc reviewed (approved):
Looks good. Tested removing the `USE_EXISTING_SRC_DIR` and it fails as expected. Thank you.
